Revisited CMAKE

fakkoweb opened the pull request:
source files are recursively found from /src
header files are recursively found from /header
build files are saved in /build (separately from project directory)
config files are stored in /cfg

Windows:
INSTALL feature - needed .dlls, media files are copied from Ogre directory and media directory into /dist according to the content of configuration files (also copied as well).
No need to copy manually any .dll into /build directory: while developing, simply add it to PATH, when installing instead all needed files are copied into /dist and all paths into cfg files are updated to reference to relative directories. The same happens from media files (media files from Ogre and from /media are copied into /dist/media and merged)

[TODO] Application tries to run looking for configuration files in installed configuration (exe in /dist). If not, it tries to find configuration files in debug/release configuration (exe in build, working dir is project's)

fakkoweb committed:
Updated for Oculus SDK 0.5
Reordered files in root. These folders will work in debug/release modes. In distribution mode (when the program is installed) all needed files are copied to /dist folder (cfg, media, dlls and exes).
On launch, program will search .cfg files in the relative path /cfg (this will be true when installed) and then in /cfg in the root. If nothing found, throw exception.
Install feature not yet implemented.

fakkoweb committed:
CmakeFile completely revisited. Lots of features added. Read description inside CMakeFile.txt and README in root and /dist directory
Moved main.cpp into /src dir (cmake recursively finds source and header files in their directories)
Readme updated
PostInstall.cmake added to parse plugins.cfg and resources.cfg files to automatically copy files from Ogre folder to /dist folder.
/dist folder added for INSTALL feature of cmake: refer to readme

fakkoweb committed:
Major program improvements:
- program can now detect if Oculus is connected or not, if not an Oculus window is created to emulate a dummy DK2 connected
- App constructor revisited to separate Ogre and Oculus initialization from Scene creation and Camera-Viewport linking
- Rift::init() and its counterpart are automatically called by Rift constructor if necessary (singleton call)
- Rift constructor can receive a NULL pointer window as parameter: Rift class will allocate to that reference a window that fits with the device connected
- More comments
- FileNotFoundException issue not solved on new Ogre::Root call: as of now you have to manually change plugins.cfg path when releasing or installing :(

fakkoweb committed:
FileNotFoundException issue solved: using ConfigFile class to get plugins.cfg first, then instantiate Ogre::root if any was found
Ogre::FileNotFoundException is thrown whether plugins.cfg or resources.cfg are not found (they can be caught by main.cpp to explain reasons to the user)

Germanunkol commented:
The CMake script does not compile for me. You mention that it should 
I see you've upgraded to the OculusSDK 0.5. I did this as well, and put the SDK into the subfolder OculusSDK. I also ran 'make' in that folder (and make install).

Setup:
Ubuntu 14.04
Oculus SDK 0.5.0.1
Ogre 1.9

Still it complains about not finding the OculusSDK. CMake output:

```
CMake Warning (dev) in CMakeLists.txt:
  A logical block opening on the line

    /home/xxx/Projects/Hiwi/OculusIntegrationTest/Source/CMakeLists.txt:125 (if)

  closes on the line

    /home/xxx/Projects/Hiwi/OculusIntegrationTest/Source/CMakeLists.txt:137 (endif)

  with mis-matching arguments.
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Error at CMakeLists.txt:136 (message):
  Failed to find module path.


-- Looking for OIS...
-- Found OIS: optimized;/usr/lib/x86_64-linux-gnu/libOIS.so;debug;/usr/lib/x86_64-linux-gnu/libOIS.so
-- Looking for OGRE...
-- Found Ogre Ghadamon (1.9.0)
-- Found OGRE: optimized;/usr/lib/x86_64-linux-gnu/libOgreMain.so;debug;/usr/lib/x86_64-linux-gnu/libOgreMain.so
-- Looking for OGRE_Paging...
-- Found OGRE_Paging: optimized;/usr/lib/x86_64-linux-gnu/libOgrePaging.so;debug;/usr/lib/x86_64-linux-gnu/libOgrePaging.so
-- Looking for OGRE_Terrain...
-- Found OGRE_Terrain: optimized;/usr/lib/x86_64-linux-gnu/libOgreTerrain.so;debug;/usr/lib/x86_64-linux-gnu/libOgreTerrain.so
-- Looking for OGRE_Property...
-- Found OGRE_Property: optimized;/usr/lib/x86_64-linux-gnu/libOgreProperty.so;debug;/usr/lib/x86_64-linux-gnu/libOgreProperty.so
-- Looking for OGRE_RTShaderSystem...
-- Found OGRE_RTShaderSystem: optimized;/usr/lib/x86_64-linux-gnu/libOgreRTShaderSystem.so;debug;/usr/lib/x86_64-linux-gnu/libOgreRTShaderSystem.so
-- Looking for OGRE_Volume...
-- Found OGRE_Volume: optimized;/usr/lib/x86_64-linux-gnu/libOgreVolume.so;debug;/usr/lib/x86_64-linux-gnu/libOgreVolume.so
-- Looking for OGRE_Overlay...
-- Found OGRE_Overlay: optimized;/usr/lib/x86_64-linux-gnu/libOgreOverlay.so;debug;/usr/lib/x86_64-linux-gnu/libOgreOverlay.so
CMake Error at CMakeLists.txt:219 (include_directories):
  include_directories given empty-string as include directory.


CMake Error at cmake/FindOculusSDK.cmake:35 (MESSAGE):
  No OCULUS_SDK_ROOT_DIR environment variable set or cmake variable
  OCULUS_SDK_ROOT_DIR set.
Call Stack (most recent call first):
  CMakeLists.txt:227 (find_package)


-- Configuring incomplete, errors occurred!
See also "/home/xxx/Projects/Hiwi/OculusIntegrationTest/BuildNew/CMakeFiles/CMakeOutput.log".
```

The complaint about line 136 - It seems that cmake lies elsewhere...?

fakkoweb commented:
Probably I removed the option of putting the OculusSDK into the same folder
of the project.
I used windows so I should improve it for linux also :) I will modify it
soon
Thank you!

2015-05-25 11:41 GMT+02:00 Germanunkol notifications@github.com:

> The CMake script does not compile for me. You mention that it should
> I see you've upgraded to the OculusSDK 0.5. I did this as well, and put
> the SDK into the subfolder OculusSDK. I also ran 'make' in that folder (and
> make install).
> Still it complains about not finding the OculusSDK. CMake output:
> 
> CMake Warning (dev) in CMakeLists.txt:
>   A logical block opening on the line
> 
> ```
> /home/xxx/Projects/Hiwi/OculusIntegrationTest/Source/CMakeLists.txt:125 (if)
> ```
> 
>   closes on the line
> 
> ```
> /home/xxx/Projects/Hiwi/OculusIntegrationTest/Source/CMakeLists.txt:137 (endif)
> ```
> 
>   with mis-matching arguments.
> This warning is for project developers.  Use -Wno-dev to suppress it.
> 
> CMake Error at CMakeLists.txt:136 (message):
>   Failed to find module path.
> 
> -- Looking for OIS...
> -- Found OIS: optimized;/usr/lib/x86_64-linux-gnu/libOIS.so;debug;/usr/lib/x86_64-linux-gnu/libOIS.so
> -- Looking for OGRE...
> -- Found Ogre Ghadamon (1.9.0)
> -- Found OGRE: optimized;/usr/lib/x86_64-linux-gnu/libOgreMain.so;debug;/usr/lib/x86_64-linux-gnu/libOgreMain.so
> -- Looking for OGRE_Paging...
> -- Found OGRE_Paging: optimized;/usr/lib/x86_64-linux-gnu/libOgrePaging.so;debug;/usr/lib/x86_64-linux-gnu/libOgrePaging.so
> -- Looking for OGRE_Terrain...
> -- Found OGRE_Terrain: optimized;/usr/lib/x86_64-linux-gnu/libOgreTerrain.so;debug;/usr/lib/x86_64-linux-gnu/libOgreTerrain.so
> -- Looking for OGRE_Property...
> -- Found OGRE_Property: optimized;/usr/lib/x86_64-linux-gnu/libOgreProperty.so;debug;/usr/lib/x86_64-linux-gnu/libOgreProperty.so
> -- Looking for OGRE_RTShaderSystem...
> -- Found OGRE_RTShaderSystem: optimized;/usr/lib/x86_64-linux-gnu/libOgreRTShaderSystem.so;debug;/usr/lib/x86_64-linux-gnu/libOgreRTShaderSystem.so
> -- Looking for OGRE_Volume...
> -- Found OGRE_Volume: optimized;/usr/lib/x86_64-linux-gnu/libOgreVolume.so;debug;/usr/lib/x86_64-linux-gnu/libOgreVolume.so
> -- Looking for OGRE_Overlay...
> -- Found OGRE_Overlay: optimized;/usr/lib/x86_64-linux-gnu/libOgreOverlay.so;debug;/usr/lib/x86_64-linux-gnu/libOgreOverlay.so
> CMake Error at CMakeLists.txt:219 (include_directories):
>   include_directories given empty-string as include directory.
> 
> CMake Error at cmake/FindOculusSDK.cmake:35 (MESSAGE):
>   No OCULUS_SDK_ROOT_DIR environment variable set or cmake variable
>   OCULUS_SDK_ROOT_DIR set.
> Call Stack (most recent call first):
>   CMakeLists.txt:227 (find_package)
> 
> -- Configuring incomplete, errors occurred!
> See also "/home/xxx/Projects/Hiwi/OculusIntegrationTest/BuildNew/CMakeFiles/CMakeOutput.log".
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/Germanunkol/OgreOculusSample/pull/1#issuecomment-105184875
> .

Germanunkol commented:
I modified my own cmake scripts to test it out with 0.5.0.1, but this didn't work for me, because of this bug/issue:
https://forums.oculus.com/viewtopic.php?f=17&t=23619&p=274160#p274160

For my cmake scripts to work, I had to tell it to also look for OVR (capital letters). It seems libovr was renamed to libOVR.
Also, the libOVR.a seems to have moved from

```
LibOVR/Lib/Linux/Release/x86_64/
```

to

```
LibOVR/Lib/Linux/x86_64/Release
```

So you might have to add that as a path to the HINTS for find_library.

(Seriously!? It's like they WANT us to spent hours debugging our makechains...!)

fakkoweb commented:
Ah! So basically it does not work even if you install OculusSDK in your
system?? Shitty
Then you should improve the OculusSDK setup section part of the makefile
and add another case... then if you can make me a pull request (or I don't
know if there is a way for me to get updates from your project)

2015-05-25 13:48 GMT+02:00 Germanunkol notifications@github.com:

> I modified my own cmake scripts to test it out with 0.5.0.1, but this
> didn't work for me, because of this bug/issue:
> https://forums.oculus.com/viewtopic.php?f=17&t=23619&p=274160#p274160
> 
> For my cmake scripts to work, I had to tell it to also look for OVR
> (capital letters). It seems libovr was renamed to libOVR.
> Also, the libOVR.a seems to have moved from
> LibOVR/Lib/Linux/Release/x86_64/
> to
> LibOVR/Lib/Linux/x86_64/Release
> 
> (Seriously!? It's like they WANT us to spent hours debugging our
> makechains...!)
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/Germanunkol/OgreOculusSample/pull/1#issuecomment-105216604
> .

fakkoweb committed:
Major improvements to cmake:
- a flag allows you to decide whether compile bundled OculusSDK or link the one in your system
- fixed issues with cmake configure/install parsing of resources config file

fakkoweb committed:
Various fixes to cmakefile

fakkoweb committed:
Huge improvement to cmakefile
UNIX still to experiment
